src/main/java/org/candlepin/subscriptions/tally/collector/RHELProductUsageCollector.java

```diff
@@ -64,4 +56,17 @@ else if (!normalizedFacts.isVirtual()) {
         }
     }
 
+    @Override
+    public void collectForHypervisor(UsageCalculation prodCalc, NormalizedFacts hypervisorFacts) {
+        int cores = hypervisorFacts.getCores() != null ? hypervisorFacts.getCores() : 0;
+        int sockets = hypervisorFacts.getSockets() != null ? hypervisorFacts.getSockets() : 0;
+
+        if (sockets == 0) {
+            throw new IllegalStateException("Hypervisor has no sockets and will not contribute to the " +
+                "totals. The tally for the RHEL product will not be accurate since all associated " +
```

Comment:
We should probably include the account number / org id here so we'd be able to identify problem areas in the logs.

Reply:
Will do.